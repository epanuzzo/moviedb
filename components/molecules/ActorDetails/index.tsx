import React from "react";

export type ActorDetailsProps = {
  name: string;
  role: string;
};

/**
 * Component to display actor details including their name and role.
 *
 * @component
 * @param {ActorDetailsProps} props - The properties for the component.
 * @param {string} props.name - The name of the actor.
 * @param {string} props.role - The role of the actor.
 * @returns {JSX.Element} A JSX element displaying the actor's initial, name, and role.
 */
const ActorDetails: React.FC<ActorDetailsProps> = ({ name, role }) => {
  return (
    <div key={name} className="flex items-center space-x-3">
      <div className="w-10 h-10 bg-gray-800 rounded-full flex items-center justify-center text-white">
        {name[0]}
      </div>
      <div>
        <p className="font-semibold">{name}</p>
        <p className="text-gray-400 text-sm">as {role}</p>
      </div>
    </div>
  );
};

export default ActorDetails;
