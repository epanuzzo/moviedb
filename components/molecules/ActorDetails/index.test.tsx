import React from "react";
import renderer from "react-test-renderer";
import ActorDetails from "./index";

describe("ActorDetails", () => {
  it("renders component", () => {
    const tree = renderer
      .create(<ActorDetails name="John" role="Actor" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
