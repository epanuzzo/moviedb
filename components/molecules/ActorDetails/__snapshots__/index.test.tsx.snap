// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorDetails renders component 1`] = `
<div
  className="flex items-center space-x-3"
>
  <div
    className="w-10 h-10 bg-gray-800 rounded-full flex items-center justify-center text-white"
  >
    J
  </div>
  <div>
    <p
      className="font-semibold"
    >
      John
    </p>
    <p
      className="text-gray-400 text-sm"
    >
      as 
      Actor
    </p>
  </div>
</div>
`;
